types/common.d.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// tslint:disable:no-any
+/**
+ * Objects with open properties
+ */
+export interface AnyObject<T = any> {
+  [property: string]: T;
+}
+
+/**
+ * Type alias for options object
+ */
+export type Options = AnyObject | null | undefined;
+
+/**
+ * Type alias for Node.js callback functions
+ */
+export type Callback<T = any> = (err: any, result?: T) => void;
+
+/**
+ * Return export type for promisified Node.js async methods.
+ *
+ * Note that juggler uses Bluebird, not the native Promise.
+ */
+export type PromiseOrVoid<T = any> = PromiseLike<T> | void;
```

Comment:
I don't see any import or definition of `PromiseLike<T>`, perhaps an oversight on your side?

I think we need to introduce some kind of a smoke test (as part of this pull request!) to at least verify that type definitions can be parsed & loaded by `tsc`.

Reply:
`PromiseLike` is a built-in type from `lib.es5.d.ts`.

Reply:
I added `npm run tsc` to validate `*.d.ts` files.

Reply:
Great!